### PR TITLE
Add CI linting and style checking

### PR DESCRIPTION
**Pull request by codeandkey**
This PR will integrate the code base with enforced style and linting checks.
It will require many changes to the code to pass the checks and won't be merged until CI passes.

Once this is merged, any subsequent PRs will be required to pass CI.

### CODE REVIEW

File: src/map.lua [hunk 1]
```diff
@@ -29,13 +29,13 @@ function map.load(name)
     assert(#map.current.tilesets <= 1)
 
     -- initialize tilesets
-    for k, v in ipairs(map.current.tilesets) do
+    for _, v in ipairs(map.current.tilesets) do
```
Comment (quigley-c) [inline src/map.lua]: Can you quickly explain this change?

Is it that 'k' is an unused variable so we replace it with a '_' to indicate this?

File: src/objects/explosion.lua [hunk 2]
```diff
@@ -6,19 +6,15 @@
 local obj = require 'obj'
 local width = 32
 local height = 32
```
Comment (quigley-c) [inline src/objects/explosion.lua]: Would it be preferable to remove the `width` variable here, as you have done in `nade.lua`?

File: src/objects/nade.lua [hunk 3]
```diff
@@ -4,17 +4,16 @@
...
 local obj = require 'obj'
-local nade_width = 16
 local nade_height = 16
 
 return {
     init = function(self)
 	    -- consts
-	    img = love.graphics.newImage('assets/sprites/16x16_nade.png')
-    	self.gravity = self.gravity or 350
+	    self.img = love.graphics.newImage('assets/sprites/16x16_nade.png')
```
Comment (quigley-c) [inline src/objects/nade.lua]: Should all variables be this way unconditionally? Can we style check for this?

File: src/map.lua (hunk 1 above)
Comment (codeandkey) [reply src/map.lua]: Yes, it's the preferred style for anonymous variables. It's then ignored by the linter as an unused.

File: src/objects/explosion.lua (hunk 2 above)
Comment (codeandkey) [reply src/objects/explosion.lua]: Both of the fields will be removed once the sprite system is merged in.

File: src/objects/nade.lua (hunk 3 above)
Comment (codeandkey) [reply src/objects/nade.lua]: Globals in object types are generally undesirable as they are shared between all objects of that type. This is good for sharing resources like images but resource caching should be implemented in another subsystem (maybe an 'assets' interface)

Comment (quigley-c) [reply src/objects/nade.lua]: To clarify, i'm referring to using `self` for instantiated object variables

**Pull request merged**